src/components/inputs/SearchInput/index.js

```diff
@@ -23,6 +38,18 @@ const SearchInput = ({
     const hasErrors = errors.length >= 1;
     // use local state to avoid re render on value prop change, avoiding special chars combinaison like "ê", "î" => IA-1432
     const [localValue, setLocalValue] = useState(value || '');
+    const onClear = () => {
+        setLocalValue('');
+    };
+    const [hasClearIcon, setClearIcon] = useState(false);
+
+    useEffect(() => {
+        if (value !== '') {
+            setClearIcon(true);
+        } else {
+            setClearIcon(false);
+        }
+    }, [value]);
```

Comment:
no need to use `useEffect` here, just create a `const` with a `useMemo` and remove `setClearIcon`